api/handlers/annotations.go

```diff
@@ -35,54 +35,110 @@ LICENSE
+}
+
+// FromAnnotation creates an AnnotationResult from a model.Annotation and key, formatting it according to the requested format.
+func FromAnnotation(annotation *model.Annotation, id int, format *api.Format) AnnotationResult {
+	var result AnnotationResult
+	if format.Requires("id") {
+		result.ID = &id
+	}
+	if format.Requires("videostream_id") {
+		result.VideoStreamID = &annotation.VideoStreamID
+	}
+	if format.Requires("timespan") {
+		result.TimeSpan = &annotation.TimeSpan
+	}
+	if format.Requires("bounding_box") {
+		result.BoundingBox = annotation.BoundingBox
+	}
+	if format.Requires("observer") {
+		result.Observer = &annotation.Observer
+	}
+	if format.Requires("observation") {
+		observation := make(map[string]string)
+		for _, o := range annotation.ObservationPairs {
+			parts := strings.Split(o, ":")
+			observation[parts[0]] = parts[1]
+		}
+		result.Observation = observation
+	}
+
+	return result
 }
 
 // GetAnnotationsQuery describes the URL query parameters required for the GetAnnotations endpoint.
 type GetAnnotationsQuery struct {
-	TimeSpan      *string `query:"timespan"`             // Optional. TODO: choose more appropriate type.
-	CaptureSource *int64  `query:"capture_source"`       // Optional.
-	Species       *string `query:"observation[species]"` // Optional.
+	TimeSpan      *string           `query:"timespan"`       // Optional. TODO: choose more appropriate type.
+	CaptureSource *int64            `query:"capture_source"` // Optional.
+	Observer      *string           `query:"observer"`       // Optional.
+	Observation   map[string]string `query:"observation"`    // Optional.
 	api.LimitAndOffset
 	api.Format
 }
 
+// CreateAnnotationBody describes the JSON format required for the CreateAnnotation endpoint.
+//
+// ID is omitted because it is chosen automatically.
+// BoundingBox is optional because some annotations might not be described by a rectangular area.
+type CreateAnnotationBody struct {
+	VideoStreamID int                `json:"videostreamId"`
+	TimeSpan      model.TimeSpan     `json:"timespan"`
+	BoundingBox   *model.BoundingBox `json:"boundingBox"` // Optional.
+	Observer      string             `json:"observer"`
+	Observation   map[string]string  `json:"observation"`
+}
+
 // GetAnnotationByID gets an annotation when provided with an ID.
 func GetAnnotationByID(ctx *fiber.Ctx) error {
-	// TODO: implement handler
+	// Parse URL.
+	format := new(api.Format)
+
+	if err := ctx.QueryParser(format); err != nil {
+		return api.InvalidRequestURL(err)
+	}
+
+	id, err := strconv.ParseInt(ctx.Params("id"), 10, 64)
+	if err != nil {
+		return api.InvalidRequestURL(err)
+	}
 
-	// id, _ := ctx.ParamsInt("id", 1)
-	return ctx.JSON("TODO")
+	// Fetch data from the datastore.
+	store := ds_client.Get()
+	key := store.IDKey("Annotation", id)
+	var annotation model.Annotation
+	if store.Get(context.Background(), key, &annotation) != nil {
+		return api.DatastoreReadFailure(err)
```

Comment:
Shouldn't "err" be from the call to store.Get in the if condition ?

Reply:
Fixed